src/orm/base_model/index.ts

```diff
@@ -721,6 +721,29 @@ class BaseModelImpl implements LucidRow {
     return this.query(options).where(key, value).firstOrFail()
   }
 
+  /**
+   * Find multiple models instance using a key/value pair
+   */
+  // @ts-expect-error - Return type should be inferred when used in a model
+  static findManyBy(clause: Record<string, unknown>, options?: ModelAdapterOptions)
+  // @ts-expect-error - Return type should be inferred when used in a model
+  static findManyBy(key: string, value: any[], options?: ModelAdapterOptions)
+  static findManyBy(
+    key: string | Record<string, unknown>,
+    value?: any[] | ModelAdapterOptions,
+    options?: ModelAdapterOptions
+  ) {
+    if (typeof key === 'object') {
+      return this.query(value as ModelAdapterOptions).where(key)
+    }
+
+    if (value === undefined) {
+      throw new Exception('"findManyBy" expects a value. Received undefined')
+    }
+
+    return this.query(options).where(key, value)
```

Comment:
I think we should execute the query here, otherwise we are returning a QueryBuilder instance and not the result of the query.

If we do not `await` the query, then the following two examples are the same

```ts
this.query().where('email', 'foo@bar.com').orderBy('id', 'desc')

this.findManyBy('email', 'foo@bar.com').orderBy('id', 'desc')
```